src/uharfbuzz/_harfbuzz.pyx

```diff
@@ -287,6 +287,23 @@ cdef class Font:
         x, y = value
         hb_font_set_scale(self._hb_font, x, y)
 
+    def variation(self, variations: Dict[str, float]) -> None:
+        cdef unsigned int size
+        cdef hb_variation_t* hb_variations
+        cdef bytes packed
+        cdef char* cstr
+        cdef hb_variation_t variation
+
+        size = len(variations)
+        hb_variations = <hb_variation_t*>malloc(size * sizeof(hb_variation_t))
+        for i, (name, value) in enumerate(variations.items()):
+            packed = name.encode()
+            variation.tag = hb_tag_from_string(packed, -1)
+            variation.value = value
+            hb_variations[i] = variation
+        hb_font_set_variations(self._hb_font, hb_variations, size)
+        free(hb_variations)
```

Comment:
maybe wrap the call to hb_font_set_variations in a `try: ... finally: free(hb_variations)` block.

Reply:
No, that’s not necessary. Actually, since we don’t expect the c++ function to raise an exception the try block is not needed here either, nevermind.

Reply:
on the other hand, the python statements in between the allocation of hb_variations and the call to free may raise exceptions, in which case we need to release it.
this is the standard pattern when using malloc/calloc/etc. from cython:
https://cython.readthedocs.io/en/latest/src/tutorial/memory_allocation.html
you'll notice that it also checks that malloc returns a NULL pointer in the unlikely event that you ran out of memory